### PR TITLE
Update to PHP 8.3 and PHPUnit 10

### PR DESCRIPTION
**Pull request by tomasnorre**
This is part of the Coordination for updating exercism PHP track to PHP 8.3 and PHPUnit 10 https://github.com/exercism/php/issues/652

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM php:8.2.7-cli-alpine3.18 AS build
+FROM php:8.3.3-cli-alpine3.18 AS build
```
Comment (tomasnorre) [inline Dockerfile:1]: I would even opt for using `php:8.3-cli-alpine3.18` instead so that we would not need to maintain on ever minor PHP update.

**Comment by mk-mxp** [comment]
As we just found out: Use `XDEBUG_MODE=off` for performance reasons.

**Comment by tomasnorre** [comment]
@mk-mxp can you reopen this PR? Or does someone else need to do it?

**Comment by mk-mxp** [comment]
I can re-open, as I just found out. But I do not have any idea of the inner workings or what might be issues with upgrading.

**Comment by tomasnorre** [comment]
> I can re-open, as I just found out. But I do not have any idea of the inner workings or what might be issues with upgrading.

Thanks. Will try to figure that out. Found that the ds php extension needed an update.

**Comment by mk-mxp** [comment]
It looks like there is a change in PHPUnit configuration needed to produce jUnit XML output in the temp directory. Or the output changed, so that there is more work for translating it to JSON for Exercism.

**Comment by tomasnorre** [comment]
> It looks like there is a change in PHPUnit configuration needed to produce jUnit XML output in the temp directory. Or the output changed, so that there is more work for translating it to JSON for Exercism.

Where did you see this? So I know what to look for. This was unknown to me. But I'm also a little on deep water here, as I don't know where and how the images are used exactly.

**Comment by mk-mxp** [comment]
> > It looks like there is a change in PHPUnit configuration needed to produce jUnit XML output in the temp directory. Or the output changed, so that there is more work for translating it to JSON for Exercism.
> 
> Where did you see this?

I scanned through the logs of the checks in the box below. Do you have rights enough to read those?

There is warnings about missing files:

```
PHP Warning:  simplexml_load_file(): I/O warning : failed to load external entity "/tmp/table-test/results.xml" in /opt/test-runner/junit-handler/src/Handler.php on line 20
```

And there is differences in JSON output like this for test `syntax-error`, which also says "Test run did not produce any output.":

```
syntax-error: comparing results.json to expected_results.json
--- /tmp/syntax-error/results.json
+++ /opt/test-runner/tests/syntax-error/expected_results.json
@@ -1 +1 @@
-{"version":3,"tests":[],"status":"error","message":"Test run did not produce any output. Check your code to see if the code exits unexpectedly before the report is generated.\n\nE.g. Using the `die` function will cause the test runner to exist unexpectedly."}
+{"version":3,"status":"fail","tests":[{"name":"testLeapYear","status":"error","test_code":"$this->assertTrue(isLeap(1996));\n","message":"LeapTest::testLeapYear\nParseError: syntax error, unexpected token \"@\", expecting \"(\"\n\n\/opt\/test-runner\/tests\/syntax-error\/Leap.php:5"},{"name":"testNonLeapYear","status":"pass","test_code":"$this->assertFalse(isLeap(1997));\n"},{"name":"testNonLeapEvenYear","status":"pass","test_code":"$this->assertFalse(isLeap(1998));\n"},{"name":"testCentury","status":"pass","test_code":"$this->assertFalse(isLeap(1900));\n"},{"name":"testFourthCentury","status":"pass","test_code":"$this->assertTrue(isLeap(2400));\n"}]}
```

**Comment by tomasnorre** [comment]
As far as I read it, the error handling of PHP Unit has changed, this chapter does not exist in previous versions of the documentation at least. https://docs.phpunit.de/en/10.5/error-handling.html

I haven't figured out what the issue is yet.

**Pull request closed without merging**